workflow/controller/operator.go

```diff
@@ -2128,6 +2128,24 @@ func (woc *wfOperationCtx) executeTemplate(ctx context.Context, nodeName string,
 		// Inject the retryAttempt number
 		localParams[common.LocalVarRetries] = strconv.Itoa(retryNum)
```

Comment:
Since we're already using `LocalVarRetries` in this scope, it does seem reasonable to me to use these other variables as well.

Reply:
Actually, I haven't looked at it long enough, but I am kind of wondering if there's a way to do this and not have to be setting these same variables twice. Should `processNodeRetries()` possibly take a `localScope` and then be able to modify it and return it to the caller? You have probably looked at the code longer than I have - what do you think?